src/logger/log_entry.py

```diff
@@ -0,0 +1,166 @@
+from datetime import datetime
+
+import marshmallow as marsh
+from peewee import CharField, DateTimeField, ForeignKeyField, IntegerField
+
+from src.base_model import BaseModel
+from src.implementor.models import Run
+
+
+class LogEntry(BaseModel):
+    """Represents a single log entry. Used to log messages from the simulation."""
+
+    class Schema(BaseModel.Schema):
+        """The marshmallow schema for the LogEntry model."""
+
+        timestamp = marsh.fields.DateTime(required=True)
+        message = marsh.fields.String(required=True)
+        run_id = marsh.fields.UUID(required=True)
+
+        def _make(self, data: dict) -> "LogEntry":
+            return LogEntry(**data)
+
+    timestamp = DateTimeField(null=False, default=datetime.now())
+
+    message = CharField(null=False)
+    run_id = ForeignKeyField(Run, null=False)
+
+
+class TrainSpawnLogEntry(LogEntry):
+    """A LogEntry that represents the spawning of a train."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainSpawnLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
```

Comment:
Please discuss with @SaturnHafen or @instant-sky if the id auf a train is an integer or uuid 😇 I'm unsure at the moment.
(Please update it in the other classes too 🥳 )

Reply:
The id is probably neither. Please discuss with @CR1337 for further details

Reply:
Will change it to a string :)